framework/docker/internal/busybox.go

```diff
@@ -21,7 +22,49 @@ var (
 	hasBusybox      bool
 )
 
-const BusyboxRef = "busybox:stable"
+const (
+	BusyboxRef = "busybox:stable"
+
+	// Retry configuration for busybox image pull
+	maxRetryAttempts  = 3
+	initialRetryDelay = 1 * time.Second
+	maxRetryDelay     = 10 * time.Second
+)
+
+// retryWithBackoff executes the given function with exponential backoff retry logic
+func retryWithBackoff(ctx context.Context, operation func() error) error {
```

Comment:
nice! This looks similar to the existing `wait.ForCondition` https://github.com/celestiaorg/tastora/blob/0162d5fd4d2fe98b182994689ec3deace7d24ccb/framework/testutil/wait/wait.go#L139

Maybe we can just use that here? Since we are only trying 3 times with a 10 second max retry delay, I don't think we gain a lot from the implementation of the exponential backoff. 

WDYT?

Reply:
thanks for that suggestion, of course it makes perfect sense, I will update it